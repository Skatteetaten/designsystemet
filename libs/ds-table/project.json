{
  "name": "ds-table",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-table/src",
  "projectType": "library",
  "tags": [],
  "implicitDependencies": ["!ds-dev-config"],
  "targets": {
    "build": {
      "executor": "@nx/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-table",
        "tsConfig": "libs/ds-table/tsconfig.lib.json",
        "project": "libs/ds-table/package.json",
        "entryFile": "libs/ds-table/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": ["libs/ds-table/rollup.config.cjs"],
        "compiler": "swc",
        "assets": [
          {
            "glob": "libs/ds-table/README.md",
            "input": ".",
            "output": "."
          },
          {
            "glob": "LICENSE",
            "input": ".",
            "output": "."
          }
        ],
        "babelUpwardRootMode": true
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/ds-table/jest.config.ts"
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-table/**/*.css",
          "libs/ds-table/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "@skatteetaten/ds-dev-config:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      },
      "dependsOn": [
        {
          "projects": "ds-core-designtokens",
          "target": "build"
        }
      ]
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-table"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish dist/libs/ds-table {args.switches}",
        "cwd": ""
      }
    }
  }
}
