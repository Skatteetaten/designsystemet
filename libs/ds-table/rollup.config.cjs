const path = require('path');

const createRollupConfig = require('../ds-dev-config/build/rollup-config-factory.cjs');

const srcDir = path.resolve(__dirname, `src`);
const pathToTsConfig = 'libs/ds-table/tsconfig.lib.json';
const pathToCSS = path.resolve(__dirname, `**/*.{css,scss}`);
const outputDir = path.resolve('./', `dist/libs/ds-table/`);

module.exports = (config) =>
  createRollupConfig(config, srcDir, pathToTsConfig, pathToCSS, outputDir);
