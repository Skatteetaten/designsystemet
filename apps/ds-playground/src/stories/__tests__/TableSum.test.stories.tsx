import { Meta, StoryObj } from '@storybook/react';
import { expect, within } from '@storybook/test';

import { Table } from '@skatteetaten/ds-table';

const meta = {
  component: Table.Sum,
  title: 'Tester/Table/Sum',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    children: { table: { disable: true } },
    colSpan: { table: { disable: true } },
    hasTopSeparator: { table: { disable: true } },
    labelAlignment: { table: { disable: true } },
    sumText: { table: { disable: true } },
    valueAlignment: { table: { disable: true } },
  },
  args: { children: '9 000 kr' },
} satisfies Meta<typeof Table.Sum>;
export default meta;
type Story = StoryObj<typeof meta>;

export const WithRef = {
  name: 'With Ref (FA1)',
  args: {
    ref: (instance: HTMLTableRowElement | null): void => {
      if (instance) {
        instance.id = 'dummyIdForwardedFromRef';
      }
    },
  },
  argTypes: {
    ref: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const table = canvas.getByRole('row');
    await expect(table).toBeInTheDocument();
    await expect(table).toHaveAttribute('id', 'dummyIdForwardedFromRef');
  },
} satisfies Story;

export const WithAttributes = {
  name: 'With Attributes (FA2-5)',
  args: {
    id: 'htmlId',
    className: 'dummyClassname',
    lang: 'nb',
    'data-testid': '123ID',
  },
  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const sum = canvas.getByTestId('123ID');
    await expect(sum).toHaveClass('dummyClassname');
    await expect(sum).toHaveAttribute('id', 'htmlId');
    await expect(sum).toHaveAttribute('lang', 'nb');
  },
} satisfies Story;

export const WithMultipleSums = {
  name: 'With Multiple Sums',
  args: {
    children: ['400 kr', '899 kr', '399 kr'],
  },
  argTypes: {
    children: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);

    await expect(canvas.getByText('400 kr')).toBeInTheDocument();
    await expect(canvas.getByText('899 kr')).toBeInTheDocument();
    await expect(canvas.getByText('399 kr')).toBeInTheDocument();
  },
} satisfies Story;
