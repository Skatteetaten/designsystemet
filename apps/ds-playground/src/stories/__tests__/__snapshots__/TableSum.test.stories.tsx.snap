// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/Table/Sum WithAttributes play-test 1`] = `
<div data-test-block="true">
  <tfoot id="htmlId"
         class="dummyClassname"
         lang="nb"
         data-testid="123ID"
  >
    <tr class>
      <th class="_dataCell_1ot0q_1 _dataCell_standard_1ot0q_1 _dataCell_right_1ot0q_23  _sum_16rlo_1"
          scope="row"
      >
        Sum
      </th>
      <td class="_dataCell_1ot0q_1 _dataCell_standard_1ot0q_1 _dataCell_right_1ot0q_23  _sum_16rlo_1">
        9 000 kr
      </td>
    </tr>
  </tfoot>
</div>
`;

exports[`Tester/Table/Sum WithMultipleSums play-test 1`] = `
<div data-test-block="true">
  <tfoot>
    <tr class>
      <th class="_dataCell_1ot0q_1 _dataCell_standard_1ot0q_1 _dataCell_right_1ot0q_23  _sum_16rlo_1"
          scope="row"
      >
        Sum
      </th>
      <td class="_dataCell_1ot0q_1 _dataCell_standard_1ot0q_1 _dataCell_right_1ot0q_23  _sum_16rlo_1">
        400 kr
      </td>
      <td class="_dataCell_1ot0q_1 _dataCell_standard_1ot0q_1 _dataCell_right_1ot0q_23  _sum_16rlo_1">
        899 kr
      </td>
      <td class="_dataCell_1ot0q_1 _dataCell_standard_1ot0q_1 _dataCell_right_1ot0q_23  _sum_16rlo_1">
        399 kr
      </td>
    </tr>
  </tfoot>
</div>
`;

exports[`Tester/Table/Sum WithRef play-test 1`] = `
<div data-test-block="true">
  <tfoot>
    <tr class
        id="dummyIdForwardedFromRef"
    >
      <th class="_dataCell_1ot0q_1 _dataCell_standard_1ot0q_1 _dataCell_right_1ot0q_23  _sum_16rlo_1"
          scope="row"
      >
        Sum
      </th>
      <td class="_dataCell_1ot0q_1 _dataCell_standard_1ot0q_1 _dataCell_right_1ot0q_23  _sum_16rlo_1">
        9 000 kr
      </td>
    </tr>
  </tfoot>
</div>
`;
